test/hotspot/jtreg/gc/TestAlwaysPreTouchBehavior.java

```diff
@@ -0,0 +1,112 @@
+ * @requires vm.gc.G1
+ * @requires os.maxMemory > 2G
+ * @library /test/lib
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run main/othervm -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -Xmx1g -Xms1g -XX:+AlwaysPreTouch gc.TestAlwaysPreTouchBehavior
+ */
+
+/**
+ * @test id=ZGenerational
+ * @summary Tests AlwaysPreTouch Behavior, pages of java heap should be pretouched with AlwaysPreTouch enabled. This test reads RSS of test process, which should be bigger than heap size(1g) with AlwaysPreTouch enabled.
+ * @requires vm.gc.ZGenerational
+ * @requires os.maxMemory > 2G
+ * @library /test/lib
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run main/othervm -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -XX:+UseZGC -XX:+ZGenerational -Xmx1g -Xms1g -XX:+AlwaysPreTouch gc.TestAlwaysPreTouchBehavior
+ */
+
+/**
+ * @test id=ZSinglegen
+ * @summary Tests AlwaysPreTouch Behavior, pages of java heap should be pretouched with AlwaysPreTouch enabled. This test reads RSS of test process, which should be bigger than heap size(1g) with AlwaysPreTouch enabled.
+ * @requires vm.gc.ZSinglegen
+ * @requires os.maxMemory > 2G
+ * @library /test/lib
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run main/othervm -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -XX:+UseZGC -XX:-ZGenerational -Xmx1g -Xms1g -XX:+AlwaysPreTouch gc.TestAlwaysPreTouchBehavior
+ */
+
+
+/**
+ * @test id=Epsilon
+ * @summary Tests AlwaysPreTouch Behavior, pages of java heap should be pretouched with AlwaysPreTouch enabled. This test reads RSS of test process, which should be bigger than heap size(1g) with AlwaysPreTouch enabled.
+ * @requires vm.gc.Epsilon
+ * @requires os.maxMemory > 2G
+ * @library /test/lib
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run main/othervm -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -XX:+UnlockExperimentalVMOptions -XX:+UseEpsilonGC -Xmx1g -Xms1g -XX:+AlwaysPreTouch gc.TestAlwaysPreTouchBehavior
+ */
+
+
+import jdk.test.lib.Asserts;
+
+import jdk.test.whitebox.WhiteBox;
+
+public class TestAlwaysPreTouchBehavior {
+
+    public static void main(String [] args) {
+    long rss = WhiteBox.getWhiteBox().rss();
+    System.out.println("RSS: " + rss);
+    if (rss == 0) {
+        System.out.println("cannot get RSS, just skip");
+        return; // Did not get available RSS, just ignore this test.
+    }
+    Runtime runtime = Runtime.getRuntime();
+    long committedMemory = runtime.totalMemory() / 1024; // in kb
```

Comment:
Why divide by KB?

Reply:
Ah yes, this is an oversight on my part. The old test converted to KB as the RSS read from `/proc/pid/status` reported KB. I had already added this conversion in the linux specific implementation of `os::rss`, so I'll go ahead and remove this.